worker.go

```diff
@@ -166,7 +166,13 @@ func ProcessLayerFromReader(datastore database.Datastore, imageFormat, name, lin
 	return files, datastore.InsertLayerComponents(layer.Name, lineage, languageComponents, files.GetRemovedFiles(), opts)
 }
 
-func detectFromFiles(files analyzer.Files, name string, parent *database.Layer, languageComponents []*component.Component, uncertifiedRHEL bool) (*database.Namespace,
+// DetectFromFiles detects the namespace and extracts the components present in
+// the files of a filesystem or image layer. For layers, the parent layer should
+// be specified. For filesystems, which don't have the concept of intermediate
+// layers, or the root layer, use `nil`. Notice that language components are not
+// extracted by DetectFromFiles, but if provided they are annotated with
```

Comment:
Language components are not detected by this function, rather they are detected by the analyzers embedded in the matcher. But we pass components here so that the certified RHEL path will [annotate the components with package info](https://github.com/stackrox/scanner/blob/346d8d3d7c017fdbad7a4ca08304204d95df33a0/pkg/analyzer/detection/detection.go#L43-L42).

Personally, I would leave that piece out of the detector, arguing that it should be its responsibility to annotate language components. But OK the way it is.